packages/firebase_auth/firebase_auth_platform_interface/lib/src/method_channel_firebase_auth.dart

```diff
@@ -0,0 +1,459 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:firebase_auth_platform_interface/firebase_auth_platform_interface.dart';
+import 'package:flutter/services.dart';
+
+import 'package:meta/meta.dart';
+
+class MethodChannelFirebaseAuth extends FirebaseAuthPlatform {
+  MethodChannelFirebaseAuth() {
+    channel.setMethodCallHandler(_callHandler);
+  }
+
+  @visibleForTesting
+  static const MethodChannel channel = MethodChannel(
+    'plugins.flutter.io/firebase_auth',
+  );
+
+  final Map<int, StreamController<PlatformUser>> _authStateChangedControllers =
+      <int, StreamController<PlatformUser>>{};
+
+  static int _nextHandle = 0;
```

Comment:
Since this is only used by the `_phoneAuthCallbacks`... `_nextPhoneAuthCallbackHandle`?

Reply:
Done.